src/list.lisp

```diff
@@ -475,3 +475,63 @@
 (defun mapcon (function list &rest more-lists)
   "Apply FUNCTION to successive CDRs of lists. Return NCONC of results."
   (map1 function (cons list more-lists) :nconc nil))
+
+
+;;; set-difference
+(defun set-difference (list1 list2 &key key (test #'eq))
+    (cond (list2
+           (let ((result '()))
+               (dolist (it list1)
+                   (when (not (member it list2 :key key :test test))
+                       (push it result)))
+               result))
+          (t list1)))
+
+;;; makeset
+(defun makeset (lst &key (test #'eq))
+    (prog ((result)
+           (seq lst))
+     feed
+       (when (null seq) (return (reverse result)))
+       (if (not (member (car seq) result :test test))
+           (setq result (cons (car seq) result)))
+       (setq seq (cdr seq))
+       (go feed)))
+
+;;; union
+(defun union (list1 list2 &key key (test #'eq))
+    (cond ((and list1 list2)
+           (let ((result (makeset list2 :test #'equal)))
+               (dolist (it (makeset list1 :test #'equal))
```

Comment:
Do you actually need `makeset` for the `dolist`? here? I don't think so. And it would probably be faster if you don't call it.

Reply:
About indentation - accepted. That's my fault.
In my standard uses lisp-body-indentation 4, but for JSCL PR I will use indent for 2 positions. Perhaps individual keywords will be with indented 1, sorry, each emacs has their own  habits.